content/doc/developer/forms/structured-form-submission.adoc

```diff
@@ -0,0 +1,190 @@
+---
+title: Structured form submission
+layout: developersection
+---
+
+== Why?
+The data model of HTML form submission is a map — each form field has a name, and when submit it, the server retrieves the values by names. This model has several problems.
+
+One is the lack of structure; if you have a repeated structure in a form, which in turn contains a nested repeated structure, just giving it a name doesn't allow you to distinguish the nesting structure. IOW, you can't tell `(A,B),(C,D)` from `(A,B,C),(D)` because the map just gives you a single list of 4 items. A similar related issue is when you put a checkbox in a repeated part of a form. If some are checked by some are not, the server can never figure out which ones are checked because that information is lost when a form is submitted.
```

Comment:
Personally I recommend "one line per sentence" formatting, to make line based feedback like this comment clearer (also making diffs easier to read).

Example:

```suggestion
One is the lack of structure; if you have a repeated structure in a form, which in turn contains a nested repeated structure, just giving it a name doesn't allow you to distinguish the nesting structure.
IOW, you can't tell `(A,B),(C,D)` from `(A,B,C),(D)` because the map just gives you a single list of 4 items.
A similar related issue is when you put a checkbox in a repeated part of a form.
If some are checked by some are not, the server can never figure out which ones are checked because that information is lost when a form is submitted.
```